### ISSUE TITLE
Reimplement maketest in terms of extract_examples?

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/AutoDoc#195]

**Comment by mohamed-barakat** [comment]
I just ran into a related problem:

The record entry `extract_examples := true` is <Q>unaware</Q> of the record entry `maketest.commands`. So with
```
gap maketest.g
```
I get no diffs, but with
```
gap tst/testall.g 
```
I get the diffs indicating that the packages list in `maketest.commands` were not loaded.

**Comment by fingolfin** [comment]
How about loading the packages in testall.g?

**Comment by sebasguts** [comment]
Yeah, that would be the right thing to do.

Previously, the `maketest.commands` were just written at the beginning of the `maketest.g`. I suggest we add the `commands` option to `extract_examples` and write them at the beginning of `testall.g`

**Comment by mohamed-barakat** [comment]
> How about loading the packages in testall.g?

Yes, I had to add them manually, but it would be nice if they are added by `maketest.commands` in an automatic way.

**Comment by fingolfin** [comment]
@mohamed-barakat If we reimplement `maketest` by using the same code base as `extract_examples`, then yes, of course we should honor `maketest.commands`. But if one is using `extract_examples`, then one should use that *instead* of `maketest` (see issue #196). So to me, it makes no sense to honor `maketest.commands` for that.

One of the various motivations for `extract_examples` is precisely that it doesn't rely on a fixed adhoc way to insert commands, as with `maketest.commands`; instead, you have full control over all commands and the order they are run, as you can simply edit the `tst.testall.g` file (and you can also call it something else, and you could have multiple files, etc. etc.). This is also IMHO much more transparent and easier to understand -- people who don't know about `maketest.commands` and its semantics don't have to puzzle over it.

tl;dr: if you use `extract_examples`, don't use `maketest`; and vice versa.

**Comment by mohamed-barakat** [comment]
I agree, and am willing to switch to `extract_examples`. I assume I can gradually start, correct?
